### PR TITLE
refactor: use bytesize::ByteSize for TrieCacheConfig

### PR DESCRIPTION
**Pull request by pugachAG**
As suggested in https://github.com/near/nearcore/pull/10409#discussion_r1450075629 to keep it consistent with the rest of `StoreConfig`.

### CODE REVIEW

File: core/store/src/trie/trie_storage.rs [hunk 1]
```diff
@@ -693,10 +694,13 @@ mod trie_cache_tests {
...
-        store_config.trie_cache.per_shard_max_bytes.insert(s0, S0_SIZE);
-        store_config.view_trie_cache.default_max_bytes = DEFAULT_VIEW_SIZE;
-        store_config.view_trie_cache.per_shard_max_bytes.insert(s0, S0_VIEW_SIZE);
+        store_config.trie_cache.default_max_bytes = bytesize::ByteSize(DEFAULT_SIZE);
+        store_config.trie_cache.per_shard_max_bytes.insert(s0, bytesize::ByteSize(S0_SIZE));
+        store_config.view_trie_cache.default_max_bytes = bytesize::ByteSize(DEFAULT_VIEW_SIZE);
+        store_config
+            .view_trie_cache
+            .per_shard_max_bytes
+            .insert(s0, bytesize::ByteSize(S0_VIEW_SIZE));
```
Comment (Ekleog-NEAR) [inline core/store/src/trie/trie_storage.rs:703]: As `ByteSize` exposes its constructor, we could have the cast already be in the `const` expressions just above :)

**Review by Ekleog-NEAR — APPROVED** [review]
LGTM, always happy to see stronger types in nearcore 🎉

**Review by wacban — APPROVED** [review]
LGTM 

Just sanity check - are there any config compatibility issues (as in would this break any existing config.json that some nodes may have deployed)?

**Comment by Ekleog-NEAR** [comment]
ByteSize [parses integers as though they were a number of bytes](https://docs.rs/bytesize/1.3.0/src/bytesize/lib.rs.html#312-321), so I don’t see any potential for compatibility issues there :)

File: core/store/src/trie/trie_storage.rs (hunk 1 above)
Comment (pugachAG) [reply core/store/src/trie/trie_storage.rs]: good point, done!

File: core/store/src/trie/config.rs [hunk 2]
```diff
@@ -8,8 +8,8 @@ use tracing::error;
 /// Default memory limit, if nothing else is configured.
 /// It is chosen to correspond roughly to the old limit, which was
 /// 50k entries * TRIE_LIMIT_CACHED_VALUE_SIZE.
-pub(crate) const DEFAULT_SHARD_CACHE_TOTAL_SIZE_LIMIT: u64 =
-    if cfg!(feature = "no_cache") { 1 } else { 50_000_000 };
+pub(crate) const DEFAULT_SHARD_CACHE_TOTAL_SIZE_LIMIT: bytesize::ByteSize =
+    if cfg!(feature = "no_cache") { bytesize::ByteSize::mb(1) } else { bytesize::ByteSize::mb(50) };
```
Comment (wacban) [inline core/store/src/trie/config.rs:12]: Is the 1B -> 1MB intentional for the no cache case?

Comment (pugachAG) [reply core/store/src/trie/config.rs:12]: ah, that was a mistake, thanks for catching!

**Pull request merged**